src/test/java/integration/CollectionWithChangingTextsTest.java

```diff
@@ -0,0 +1,64 @@
+package integration;
+
+import com.codeborne.selenide.SelenideElement;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+
+import static com.codeborne.selenide.CollectionCondition.size;
+import static com.codeborne.selenide.Condition.text;
+import static java.lang.Thread.sleep;
+import static org.assertj.core.api.Assertions.assertThat;
+
+final class CollectionWithChangingTextsTest extends ITest {
+  @BeforeEach
+  void openTestPage() {
+    openFile("collection_with_changing_texts.html");
+    setTimeout(4000);
+  }
+
+  @Test
+  void snapshotDoesNotReloadElements() {
+    Iterator<SelenideElement> it = $$("#collection li").snapshot()
+      .shouldHave(size(3))
+      .iterator();
+    it.next().shouldHave(text("Item #1"));
+    it.next().shouldHave(text("Item #2"));
+    it.next().shouldHave(text("Item #3"));
+    assertThat(it.hasNext()).isFalse();
+  }
+
+  @Test
+  void iteratorReloadsCollectionOnEveryCall() {
+    Iterator<SelenideElement> it = $$("#collection li").iterator();
+    it.next().shouldHave(text("Item #1"));
+    it.next().shouldHave(text("Item #2"));
+    it.next().shouldHave(text("Item #3"));
+    assertThat(it.hasNext()).isFalse();
+  }
+
+  @Test
+  void reloadingIterable_reloadsCollectionOnEveryCall() throws InterruptedException {
+    Iterator<SelenideElement> it = $$("#collection li").asReloadingIterable().iterator();
+    it.next().shouldHave(text("Item #1"));
+    it.next().shouldHave(text("Item #2"));
+    it.next().shouldHave(text("Item #3"));
+    sleep(350);
```

Comment:
IMO, this looks a bit "magical", implicit.

Maybe it's better to leave a comment here that the page has a script that adds additional elements to the collection after this timeout?

Reply:
yes, I will add a comment. 
Though, many of Selenide tests use such approach: an html page with dynamic JS inside and _a Java test.